### DISCUSSION TITLE
About `el_client_type` customizations

### DISCUSSION THREAD

**Discussion by seaona** [ethpandaops/ethereum-package#511] [Q&A] [answered]
hi there :wave: I have a couple of questions around the EL client customization. It's my first time setting up the environment, so apologies in advanced, if I missed something on the documentation :bowing_woman: 

On the README it states that:

> The Docker image that should be used for the EL client; leave blank to use the default for the client type
>   Defaults by client:
>   - geth: ethereum/client-go:latest

I wanted to know:

1. What does `latest` refer to here? Does it mean, the latest **[released version](https://github.com/ethereum/go-ethereum/releases)**? Or does it mean, what's currently in the `master` branch?
2. I tried to change the `el_client_type` value to a specific geth version like this `ethereum/client-go:v1.13.10`, however I see the error `fail: ethereum/client-go:v1.13.10 received an empty image name and we don't have a default for it`. Is there something else I need to do in order to setup a custom version for geth?
3. Following up on question 2, if I want to set my local geth build, how should I proceed on doing so? I imagine I would need to create a Docker image of my build and somehow pass it to the yaml config? :thinking:

Thank you very much in advance and apologies again if those questions are already solved somewhere else in the docs :pray:

**Comment by barnabasbusa** [comment, marked as answer]
`latest` will refer to the reference of their dockerhub image, which currently is: https://hub.docker.com/layers/ethereum/client-go/latest/images/sha256-e3a568edc60d57261bdd774c852fd20b0a6ac347354020d9eb8a0fdd6e31b091?context=explore
`latest` will mean different things for different projects, as far as I know geth's latest dockerhub image is automatically built/published based on the latest master branch. 

`el_client_type` can be geth/nethermind/reth ...
what you looking for is `el_client_image` field, where you can specify which image you plan to use. 

However, we currently have a pretty big refactoring PR which will change these names (once the pr gets merged). https://github.com/kurtosis-tech/ethereum-package/pull/508

3. You can build a local docker image, tag it with something like testing-geth-image:latest, then for the el_client_image you specify that image name. Kurtosis will try to find the image locally first, and if it can't find it, it will try to fetch it from dockerhub.

**Reply by seaona** [reply]
Thank you so much @barnabasbusa :heart_eyes: this has been really helpful and clarifies all the questions I had.
I'll keep an eye on the PR, for the name change :+1:
